.github/workflows/settings-2-deploy.yml

```diff
@@ -0,0 +1,86 @@
+name: Config Settings Deployment
+on:
+  workflow_call:
+    inputs:
+      deployment-environment:
+        type: string
+        required: true
+        description: GitHub Deployment Environment Name
+      spack-type:
+        type: string
+        required: true
+        description: The type of spack deployment (eg. Prerelease, Release)
+env:
+  CONFIG_SETTINGS_PATH: ./config/settings.json
+jobs:
+  update-environment:
+    name: Update ${{ inputs.deployment-environment }} ${{ inputs.spack-type }} Settings
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.deployment-environment }} ${{ inputs.spack-type }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup spack updates
+        id: spack
+        # TODO: Since we can't format this json as an input for this job (see the earlier workflow) we need to do it here
+        run: |
+          updates=$(jq --compact-output --raw-output \
+            --arg env "${{ inputs.deployment-environment }}" \
+            --arg type "${{ inputs.spack-type }}" \
+            '.deployment[$env][$type].spack' \
+            ${{ env.CONFIG_SETTINGS_PATH }}
+          )
+          versions=$(jq --compact-output --raw-output \
+            --arg env "${{ inputs.deployment-environment }}" \
+            --arg type "${{ inputs.spack-type }}" \
+            '.deployment[$env][$type].spack | keys | @sh' \
+            ${{ env.CONFIG_SETTINGS_PATH }}
+          )
+
+          echo "$updates"
+          echo "$versions"
+
+          echo "updates=$updates" >> $GITHUB_OUTPUT
+          echo "versions=$versions" >> $GITHUB_OUTPUT
+
+      - name: Setup SSH
+        id: ssh
+        uses: access-nri/actions/.github/actions/setup-ssh@main
+        with:
+          hosts: ${{ secrets.HOST }}
+          private-key: ${{ secrets.SSH_KEY }}
+
+      - name: Update
+        continue-on-error: true
```

Comment:
What errors do we expect to need to continue on/from?

Reply:
https://github.com/ACCESS-NRI/build-cd/pull/111/files/a34bb0b53657f169450c8ac31a9c3463c9837567#diff-e5e653c61b96bdcaedd9c75e01adfcd8b5dde47f5e56329feb359c6759ae26bdR66-R67 mostly